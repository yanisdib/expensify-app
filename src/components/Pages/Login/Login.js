import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../../../actions/auth';

export const Login = ({ startLogin }) => (
    <section id="login">
        <h1>Expense Manager</h1>
        <h5>A simple expense manager</h5>
        <br />
        <p>Please login to access your expenses.</p>
        <br />
        <button className="btn btn-primary" onClick={startLogin}>
            <img src="/media/images/google_logo_16x16.png" alt="Google logo"></img>
             Login with Google account ->
        </button>
    </section>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(Login);