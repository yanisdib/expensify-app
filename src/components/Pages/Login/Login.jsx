import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../../../actions/auth';

export const Login = ({ startLogin }) => (
    <div id="login">
        Please login to go any further.
        <br />
        <button onClick={startLogin}>Login</button>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(Login);