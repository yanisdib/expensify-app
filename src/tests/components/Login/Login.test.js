import React from 'react';
import { shallow } from 'enzyme';
import { Login } from '../../../components/Pages/Login/Login';
let startLogin, wrapper;

// Create a spy
// Pass the spy
beforeEach(() => {
    startLogin = jest.fn();
    wrapper = shallow(<Login startLogin={startLogin} />);
});

test('should render Login component correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call startLogin on button click', () => {
    // Simulate a click event
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});