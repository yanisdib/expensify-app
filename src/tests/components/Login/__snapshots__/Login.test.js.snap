// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login component correctly 1`] = `
<div
  id="login"
>
  Please login to go any further.
  <br />
  <button
    onClick={[MockFunction]}
  >
    Login
  </button>
</div>
`;
